src/meson.build

```diff
@@ -0,0 +1,95 @@
+# main sources
+core_sources_c = [ 'src/main.c',
+                   'src/profanity.c',
```

Comment:
Instead of making an array of strings, and including the current directory (`src/`) you can use:

```meson
core_sources_c = files(
    'main.c',
    'profanity.c',
    ...
)
```

The returned File object automatically includes the leading `src/` for you.

Reply:
what about the files that also belong to the core sources like `src/xmpp/contact.c`? Can I add them via `xmpp/contact.c` or will I need a src/xmpp/meson.build file for this?

Reply:
The former.

```meson
core_sources_c = files(
    'main.c',
    'profanity.c',
    'xmpp/contact.c',
    'event/server_events.c',
    'ui/window.c',
    'command/cmd_defs.c',
    'config/accounts.c',
    ...
)
```